src/out/OutSimPack2.ts

```diff
@@ -0,0 +1,101 @@
+import {
+  array,
+  byte,
+  float,
+  int,
+  string,
+  struct,
+  unsigned,
+} from '../decorators';
+import { Struct } from '../packets';
+import { OutSimInputs } from './OutSimInputs';
+import { OutSimMain } from './OutSimMain';
+import { OutSimOptions } from './OutSimOptions';
+import { OutSimWheel } from './OutSimWheel';
+
+export class OutSimPack2 extends Struct {
+  // if (OSOpts & OSO_HEADER)
+  @string(4) Header = '';
+  // if (OSOpts & OSO_ID)
+  @int() ID = 0;
+  // if (OSOpts & OSO_TIME)
+  @unsigned() Time = 0;
+  // if (OSOpts & OSO_MAIN)
+  @struct(OutSimMain) OSMain = new OutSimMain();
```

Comment:
I'm thinking if it would be better to place the subsection properties flat in the `OutSimPack2` struct without creating separate sub-structs, just like you did it for the `OSO_DRIVE` section.

Then in the app you would access all properties like `outSimPack2.AngVel` instead of `outSimPack2.OSMain.AngVel` and so on.

What do you think?

Reply:
This structure is as described in `LFS/docs/OutSimPack.txt`. 
I didn't look much into what is contained in each struct property, but I assume the properties are extracted to separate structures because otherwise the names could be ambiguous, i.e.:
`OSMain.AngVel` and `OSWheel[x].AngVel`.
I suppose the user could remap the data onto more convenient model if they desire at their end.
If the structure was written by me I would have also made another OSDrive structure :)

Reply:
I see, it makes sense. Let's keep it then 🙂